README.md

```diff
@@ -238,76 +238,119 @@ the DRAM implementation. However, here as well we will encounter are a few thing
-spine (bag of batches) to the disk as a separate file and ensure that it has been fully written out to disk (by 
-writing to a temporary file first + fsync + rename with old file which is atomic). This avoids the problem of having 
-partially written spine meta-data. However, this is not enough since we have many spines in a dbsp program, and we 
-need to have the entire dbsp state (all spines) consistent. 
-
-There are different ways to solve this problem:
-
-##### With a log
-
-- The modifications to spines happen during a transaction.
-- Whenever we add or remove a batch to a spine, we also log it as an entry to an undo log.
-- When we commit, we clear the undo log, and in addition we can remove any batch files we now no longer need (e.g., 
-  because they got merged into a new badge).
-- If we need to rollback/or recover after crash we read the undo log and undo all changes to the spines (e.g., remove 
-  batches that got added and add batches that got removed).
-
-##### Using Timestamps
-
-- The modifications to spines happen during a transaction
-- We store the `earliest` and `last` timestamp (e.g., incrementing transaction id) for each batch
-- We set `earliest` of every incoming batch to the current timestamp (and `last` to nil)
-- on a merge, the timestamp of the new batch becomes `earliest` = current timestamp
-- the timestamp of the older two batches becomes `last` = current timestamp-1
-- When we commit, we store the timestamp of the commit as a checkpoint
-- When we need to rollback/or recover after a crash we read timestamp for each batch in a spine and remove all batches 
-  that have `earliest` timestamp larger than the checkpoint timestamp. We add back all batches where last == checkpoint.
-- We can remove all batches where `last` is smaller than the last checkpoint we need
-- We also need to make sure we don't merge any batches across commits.
-
-This is similar to the log approach, but we don't need to store the undo log as we can just use the timestamps to
-figure out which batches to remove. While we can write e.g., earliest as part of writing the batch, I'm not sure
-how/where we write the `last` timestamp (@Leonid).
-
-One of the control plane APIs will be a transaction API to ensure consistency of all DBSP Spine (and Layer) state. 
-Another API will be added to restore to a previous checkpoint.
+Persistence is exposed through a control plane API added to the circuit handle. The control plane API involves:
+
+- providing a storage location (directory)
+- an API call for taking a checkpoint of the circuit
+- an API call to delete (the oldest available) checkpoint
+- a configuration option when initializing the circuit to resume from a given storage location and checkpoint
+
+Next we will explain these operations in more detail.
+
+#### Providing a storage location
+
+The circuit will store its persistent state in a set of files and sub-directories in a user-provided path.
```

Comment:
"user" is unclear. Perhaps "Specified". And this path is really a root directory.